fe/fe-core/src/main/java/com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java

```diff
@@ -311,6 +312,8 @@ public Void visitCreateMaterializedViewStatement(CreateMaterializedViewStatement
                 Preconditions.checkState(pair.second < outputExpressions.size());
                 columnExprMap.put(pair.first, outputExpressions.get(pair.second));
             }
+
+            checkBaseTablePartitionType(aliasTableMap);
```

Comment:
Only check partition type for partitioned mv since unpartitioned mv for list partitioned base tables is supported?

```
 if (statement.getPartitionExpDesc() != null) {
               checkBaseTablePartitionType(aliasTableMap);
```